### DISCUSSION TITLE
Debugging Extension

### DISCUSSION THREAD

**Discussion by GoldenGamerLP** [Minestom/Minestom#1070] [Help] [answered]
Quick question: Is it possible to debug my extension on performance? And if yes, how?

**Comment by KrystilizeNevaDies** [comment]
Yes, you can use any java profiling tool. e.g. intellij async profiler, jprofiler

**Comment by GoldenGamerLP** [comment, marked as answer]
Can you explain how? especially with the extension

**Reply by ryleu** [reply]
please use a reply, not another answer

**Reply by ZakShearman** [reply]
There shouldn't be any difference to profiling with another Java app within your IDE.

This is just an example of extremely basic usage of JFR within intellij, there are much more extensive guides online:

https://user-images.githubusercontent.com/34372536/168428208-3b3c06e5-e585-4303-a483-8b75a40029e1.mp4

Bear in mind GitHub has a reply feature, you do not need to suggest a new answer to respond 😃

**Reply by GoldenGamerLP** [reply]
Thank you :)
